spyder/plugins/editor/widgets/tests/fixtures.py

```diff
@@ -68,19 +47,42 @@ def setup_editor(qtbot):
     return editorStack, finfo.editor
 
 
-@pytest.fixture
-def lsp_codeeditor(qtbot):
-    """CodeEditor instance with LSP services activated."""
+@pytest.fixture(scope="module")
+def qtbot_module(qapp, request):
+    """Module fixture for qtbot."""
+    result = QtBot(request)
+    return result
+
+
+@pytest.fixture(scope="module")
+def lsp_manager(qtbot_module):
+    """Create an LSP manager instance."""
     # Activate pycodestyle and pydocstyle
     CONF.set('lsp-server', 'pycodestyle', True)
     CONF.set('lsp-server', 'pydocstyle', True)
 
-    # Tell CodeEditor to use introspection
+    # Create the manager
     os.environ['SPY_TEST_USE_INTROSPECTION'] = 'True'
+    manager = LSPManager(parent=None)
 
-    # Create an LSPManager instance to be able to start an LSP client
-    lsp_manager = LSPManager(parent=None)
+    with qtbot_module.waitSignal(manager.sig_initialize, timeout=30000) as blocker:
+        manager.start_client('python')
 
+    settings, language = blocker.args
+    manager.clients[language]['server_settings'] = settings
+
+    yield manager
+
+    # Tear down operations
+    manager.shutdown()
+    os.environ['SPY_TEST_USE_INTROSPECTION'] = 'False'
+    CONF.set('lsp-server', 'pycodestyle', False)
+    CONF.set('lsp-server', 'pydocstyle', False)
```

Comment:
Probably work for another PR, but I suggest we run the tests in a separate config that should be reset to defaults after each test (if using such fixture) or one that does not reset for module level tests.

Reply:
I agree, but I don't know how we could do it.